### DISCUSSION TITLE
[@types/three] AnimationClip.findByName is incorrectly typed

### DISCUSSION THREAD

**Discussion by jackbonaguro** [DefinitelyTyped/DefinitelyTyped#71383] [Issues with a @types package]
Source Permalink: https://github.com/mrdoob/three.js/blob/f5f25c3e9b814b001c51958effeb37da495f03a3/src/animation/AnimationClip.js#L119

Type Permalink: https://github.com/jackbonaguro/DefinitelyTyped/blob/a7a0c867fd449e601bcd8b2c3ed3bfeaa4b152e6/types/three/src/animation/AnimationClip.d.ts#L42

As you can see, the `clipArray` parameter should actually be `objectOrClipArray`, and it can be an array of clips OR an object with the `geometry` property (likely a `Mesh`) OR an object with the `animations` property (likely an `Object3D`).

[a comment by an automated account is not shown]

**Comment by jackbonaguro** [comment]
I've contributed to @types/three before but it seems more demanding now. I made a fix here but don't want to mess up the PR process since there's a lot of steps: https://github.com/jackbonaguro/DefinitelyTyped/tree/fixFindByNameType

**Reply by Methuselah96** [reply]
Thanks for the fix, I've included it [here](https://github.com/three-types/three-ts-types/pull/1397), which will be included in the next release. Let me know if I can do anything to make the contribution process easier!
